File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res(n + 1);
        for (int i = 0; i <= n; i++)
        {
            int num = i;
            while (num)
            {
                num &= (num - 1);
                res[i]++;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = [[]] # [[], [1], [2], [1, 2], [3], [1, 3], [2, 3], [1, 2, 3]]

        for num in nums:
            res += [subset + [num] for subset in res]

        return res
        